File: frontend-server/src/app/services/http/united/common-data.service.ts
import { Observable } from "rxjs";
import { CommonDataDTO } from "../../../model/dto/united/common-data-dto";
import { Injectable } from "@angular/core";
import { HttpClient } from "@angular/common/http";
import { RestConfig } from "../../../rest.config";

@Injectable({
  providedIn: 'root'
})
export class CommonDataService {

  constructor(private httpClient: HttpClient,
              private config: RestConfig){
  }
  
  getByEnterpriseIdAndDepartmentId(enterpriseId: number,
                                   departmentId: number): Observable<CommonDataDTO> {
    return this.httpClient.get<CommonDataDTO>(
      `${this.config.baseUrl}/enterprises/${enterpriseId}/departments/${departmentId}/common_data`
    );
  }
}

File: work-schedule-service/src/main/java/com/averude/uksatse/scheduler/microservice/workschedule/service/united/CommonDataService.java
package com.averude.uksatse.scheduler.microservice.workschedule.service.united;

import com.averude.uksatse.scheduler.microservice.workschedule.model.CommonDataDTO;

public interface CommonDataService {
    CommonDataDTO getByEnterpriseIdAndDepartmentId(Long enterpriseId, Long departmentId);
}
